casper/src/main/rholang/MakePoS.rho

```diff
@@ -183,6 +186,48 @@ in {
           }
         } |
       
+        //TODO: Fill in this contract (RHOL-462)
+        contract this(@"distribute", @purse) = {
+          Nil
+        } |
+      
+        paymentStore!((Nil, Nil, Nil, Nil)) |
+      
+        //Make it easy to query the last payment so that validators
+        //can check how much phlo there is for the deploy.
+        contract this(@"lastPayment", return) = {
+          for(@payment <- paymentStore) {
+            paymentStore!(payment) |
+            return!(payment)
+          }
+        } |
+      
+        //This is the contract that deploy payment code ultimately needs
+        //to call in order to pay for the deploy.
+        contract this(@"pay", @purse, return) = {
```

Comment:
Who is going to validate that the payment is being done for valid deploy? I thought that the purpose of deploy parameters was to disallow substitution of deployed code with validator's one. You're not making any use of the deploy parameters other than putting them into the `paymentStore`.

Shouldn't wallet/purse be checking that it's requested to make a transfer for the code deployed from the owner of it?

Reply:
The purpose of this API is to simply give money to the validators. My model for how the whole process would go is that the user's "payment code" that gets run on the "short leash" handles the verification of things like the code hash and signature. Then as a final step it calls this API to actually pay the validators.

Reply:
OK, apparently I misunderstood how the short leash will work.